torchgeo/datamodules/utils.py

```diff
@@ -52,3 +55,75 @@ def dataset_split(
             [train_length, val_length, test_length],
             generator=Generator().manual_seed(0),
         )
+
+
+def group_shuffle_split(
+    groups: Iterable[Any],
```

Comment:
```suggestion
    groups: Iterable[object],
```
Any means disable type checking, object means must support all possible input types. See if this passes. If not, revert it.